### ISSUE TITLE
Alert - a11y documentation for announcing alerts (role="alert")

### ISSUE THREAD

**Issue by humancompanion-usds** [department-of-veterans-affairs/vets-design-system-documentation#2145]
## Description
Update a11y documentation and guidance for Alert on design.va.gov. Brian on Governance recently "clarified" use of `role="alert"` on Alerts. This guidance needs to be added to the Alert component to make it clear to teams how and when Alerts should be announced. [Slack thread](https://dsva.slack.com/archives/C023ZKDJJK1/p1696016693905639?thread_ts=1693943715.840939&cid=C023ZKDJJK1). This thread needs to be boiled down into a decision tree that folks can follow to arrive at a sensible conclusion. If focus management should also be covered then let's also create a sub-section for that. 

## Details
The guidance should be reviewed by Brian so that he agrees to not contradict guidance during collab touchpoints. 

## Tasks
- [ ] Write component documentation for design.va.gov
- [ ] Request documentation review

## Acceptance Criteria
- [ ] Component a11y documentation is update on design.va.gov
